test/Microsoft.Azure.Devices.Edge.Test.Common/EdgeConfiguration.cs

```diff
@@ -61,39 +179,40 @@ public void AddRegistryCredentials(string address, string username, string passw
                 });
         }
 
-        // Adds proxy information to each module in Edge Agent's desired properties. Call this
-        // method after you've added all the modules that need proxy information.
-        public void AddProxy(Uri proxy)
+        public Module AddEdgeAgent(string image)
         {
-            this.ForEachModule(
-                (name, module) =>
+            const string Name = "edgeAgent";
+
+            this.UpdateAgentDesiredProperties(
+                desired =>
                 {
-                    JObject env = GetOrAddObject("env", module);
-                    env.Add(
-                        "https_proxy",
-                        JToken.FromObject(
-                            new
-                            {
-                                value = proxy.ToString()
-                            }));
-                    env.Add(
-                        "UpstreamProtocol",
+                    JObject systemModules = GetOrAddObject("systemModules", desired);
+                    systemModules.Add(
+                        Name,
                         JToken.FromObject(
                             new
                             {
-                                value = UpstreamProtocolType.AmqpWs.ToString()
+                                type = "docker",
+                                settings = new
+                                {
+                                    image
+                                }
                             }));
                 });
+
+            return new Module(Name, this);
         }
 
-        public void AddEdgeHub(string image)
+        public Module AddEdgeHub(string image)
         {
+            const string Name = "edgeHub";
+
             this.UpdateAgentDesiredProperties(
                 desired =>
                 {
-                    JObject systemModules = desired.Get<JObject>("systemModules");
+                    JObject systemModules = GetOrAddObject("systemModules", desired);
```

Comment:
Do we need to lock in GetOrAddObject?

Reply:
I don't think it's used by multiple threads, but still need to check...

Reply:
`GetOrAddObject` no longer exists...